### ISSUE TITLE
Module Can Cause System Build Failures Even If Not Enabled

### ISSUE THREAD

**Issue by Infinidoge** [nix-community/NixOS-WSL#152]
## Bug description
When the module has been imported and no options are set, a rebuild can fail with "error: attribute 'nixos' missing".
This is due to https://github.com/nix-community/NixOS-WSL/commit/27679db0469bc51ce4d0b334c74c4b0642d62009 causing `cfg.defaultUser` to be eagerly evaluated regardless of `cfg.enable` being false/the default.
This causes a rebuild failure when the user `nixos` doesn't exist.

## To Reproduce
Import module and set no options.
Experience rebuild failure

**Expected behavior**
No rebuild failure

## Logs
```
error: attribute 'nixos' missing

       at /nix/store/pqfb6jic287q3d7s9jvz1w8lgrjnm3bi-source/modules/wsl-distro.nix:40:23:

           39|         inherit (cfg) automountPath;
           40|         defaultUser = config.users.users.${cfg.defaultUser};
             |                       ^
           41|       };

       … while evaluating the attribute 'uid' of the derivation 'syschdemd'
```

## Additional Context

For simplicity, I have modules imported by all of the NixOS configurations I have in my configuration flake, and was confused to find that after bumping NixOS-WSL, the rebuild on my very-not-Windows laptop failed with a WSL-related issue.

The best solution would be to just not import the WSL module unless I absolutely need it, or to set `defaultUser`, but ideally the module wouldn't cause any problems by merely being imported, only when actually enabled.

**Comment by SuperSandro2000** [comment]
Fixed by #151

**Comment by nzbr** [comment]
This wasn't supposed to happen, I'm very sorry for that!
